website/docs/automation/gitops/flux/ci.md

```diff
@@ -0,0 +1,231 @@
+
+```file
+manifests/modules/automation/gitops/flux/imagerepository.yaml
+```
+
+2. ImagePolicy:
+
+```file
+manifests/modules/automation/gitops/flux/imagepolicy.yaml
+```
+
+3. ImageUpdateAutomation:
+
+```file
+manifests/modules/automation/gitops/flux/imageupdateautomation.yaml
+```
+
+```bash
+$ cp ~/environment/eks-workshop/modules/automation/gitops/flux/image*.yaml ~/environment/retail-store-sample-codecommit/
+$ yq -i ".spec.image = env(IMAGE_URI_UI)" ~/environment/retail-store-sample-codecommit/imagerepository.yaml
+$ less ~/environment/retail-store-sample-codecommit/imagerepository.yaml | grep image:
+$ kubectl apply -f ~/environment/retail-store-sample-codecommit/imagerepository.yaml
+$ kubectl apply -f ~/environment/retail-store-sample-codecommit/imagepolicy.yaml
+$ kubectl apply -f ~/environment/retail-store-sample-codecommit/imageupdateautomation.yaml
+```
+
+We created the following architecture:
+
+![ci-eks-gitops](assets/ci-eks-gitops.png)
+
+Now, lets reconcile the changes.
+
+```bash
+$ flux reconcile image repository ui
+$ flux reconcile source git flux-system
+$ flux reconcile kustomization apps
+$ kubectl wait deployment -n ui ui --for condition=Available=True --timeout=120s
+$ git -C ~/environment/flux pull
+$ kubectl -n ui get pods
+```
+
+We can check that `image:` in the `deployment` has been updated to a new tag.
+
+```bash
+$ kubectl -n ui describe deployment ui | grep Image
+```
+
+To access `UI` using a browser we need to expose it using `Ingress` and `Load Balancer`.
+
+Let's create an Ingress resource with the following manifest:
+
+```file
+manifests/modules/exposing/ingress/creating-ingress/ingress.yaml
+```
+
+This will cause the AWS Load Balancer Controller to provision an Application Load Balancer and configure it to route traffic to the Pods for the `ui` application.
+
+```bash timeout=180
+$ kubectl apply -k ~/environment/eks-workshop/modules/exposing/ingress/creating-ingress
+```
+
+Let's inspect the Ingress object created:
+
+```bash
+$ kubectl get ingress ui -n ui
+NAME   CLASS   HOSTS   ADDRESS                                            PORTS   AGE
+ui     alb     *       k8s-ui-ui-1268651632.us-west-2.elb.amazonaws.com   80      15s
+```
+
+We wait 2-5 minutes until Application Load Balancer will be provisioned and check the UI page using url of the ingress.
+
+```bash test=false
+$ export UI_URL=$(kubectl get ingress -n ui ui -o jsonpath="{.status.loadBalancer.ingress[*].hostname}{'\n'}")
+$ while [[ $(curl -s -o /dev/null -w "%{http_code}" $UI_URL/home) != "200" ]]; do sleep 1; done
+$ curl $UI_URL/home | grep "Retail Store Sample"
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+        <title>Retail Store Sample App</title>
+100                <a class="navbar-brand" href="/home">Retail Store Sample</a>
+ 13254  100 13254    0     0   236k      0 --:--:-- --:--:-- --:--:--  239k
```

Comment:
```suggestion
$ wait-for-lb $UI_URL
```

There is already a `wait-for-lb` to make it simpler and cleaner to wait for url to be up